### PR TITLE
Added ability to output extension info to clipboard

### PR DESCRIPTION
**Pull request by lannonbr**
Implements #47296

I added an action to the extensions actions cog to copy a chunk of metadata from an extension to the clipboard.

The copied text includes: id, description, version, publisher, and marketplace link.

This also could be useful so if a user has a bug with an extension, they could copy this into an issue and the extension maintainer could know some metadata like the extension version when they're trying to debug an issue.

### CODE REVIEW

File: src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts [hunk 1]
```diff
@@ -872,6 +876,39 @@ export class ManageExtensionAction extends Action {
 	}
 }
 
+export class ExtensionInfoAction extends Action implements IExtensionAction {
+
+	static readonly ID = 'extensions.extensionInfo';
+	static LABEL = localize('extensionInfoAction', "Copy Extension info to clipboard");
```
Comment (ramya-rao-a) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: The `to clipboard` here is unnecessary. We have other copy actions in the product where it is assumed that the user knows that the text is copied to clipboard.

How about `Copy Extension information`?

File: src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts [hunk 2]
```diff
@@ -872,6 +876,39 @@ export class ManageExtensionAction extends Action {
...
+		const localizedExtension = localize('extensionInfoId', 'Extension') + ': ' + id;
+		const localizedDescription = localize('extensionInfoDescription', 'Description') + ': ' + description;
+		const localizedVersion = localize('extensionInfoVersion', 'Version') + ': ' + version;
+		const localizedPublisher = localize('extensionInfoPublisher', 'Publisher') + ': ' + publisherDisplayName;
+		const localizedVSMarketplaceLink = localize('extensionInfoVSMarketplaceLink', 'VS Marketplace Link') + ': https://marketplace.visualstudio.com/items?itemName=' + id;
+
+		const clipboardStr = `${localizedExtension}\n${localizedDescription}\n${localizedVersion}\n${localizedPublisher}\n${localizedVSMarketplaceLink}`;
+
+		clipboard.writeText(clipboardStr);
+		this.notificationService.info(localize('extensionInfoActionNotification', 'Extension info copied to clipboard'));
```
Comment (ramya-rao-a) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: To stay consistent with other copy actions in the product that do not show any notifications on copy (For eg: `Copy Path`), let's skip the notification here.

File: src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts (hunk 1 above)
Comment (lannonbr) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: Done

File: src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts (hunk 2 above)
Comment (lannonbr) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: Done

**Comment by ramya-rao-a** [comment]
@lannonbr This PR is currently in a very bad state most probably due to a bad merge from master. I'll close this PR. Can you create a new one?

**Pull request closed without merging**